src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesSlave.java

```diff
@@ -207,24 +248,19 @@ protected void _terminate(TaskListener listener) throws IOException, Interrupted
             listener.fatalError(msg);
             return;
         }
-        KubernetesCloud cloud;
-        try {
-            cloud = getKubernetesCloud();
-        } catch (IllegalStateException e) {
-            e.printStackTrace(listener.fatalError("Unable to terminate agent. Cloud may have been removed. There may be leftover resources on the Kubernetes cluster."));
-            LOGGER.log(Level.SEVERE, String.format("Unable to terminate agent %s. Cloud may have been removed. There may be leftover resources on the Kubernetes cluster.", name));
-            return;
-        }
-        KubernetesClient client;
-        try {
-            client = cloud.connect();
-        } catch (UnrecoverableKeyException | CertificateEncodingException | NoSuchAlgorithmException
-                | KeyStoreException e) {
-            String msg = String.format("Failed to connect to cloud %s", getCloudName());
-            e.printStackTrace(listener.fatalError(msg));
-            return;
+
+        if (deletePod) {
+            deleteSlavePod(listener, client);
+        } else {
+            LOGGER.log(Level.WARNING, "Slave pod {0} was not deleted due to retention policy {1}.",
+                    new Object[] { name, getPodRetention(cloud) });
```

Comment:
Log a warning if the slave pod was not deleted, as this may mean a running pod is still active and consuming resources.